src/convert.rs

```diff
@@ -534,6 +594,167 @@ pub fn elf_to_tbf<W: Write>(
     output.write_all(&rel_data_len)?;
     output.write_all(relocation_binary.as_ref())?;
 
+    // That is everything that we are going to include in the app binary
+    // that is covered by integrity. Now add footers.
+
+    let footers_len = total_size - tbfheader.binary_end_offset() as usize;
+    let mut footer_space_remaining = footers_len;
+    if sha256 {
+        // Total length
+        let sha256_len = mem::size_of::<header::TbfHeaderTlv>()
+            + mem::size_of::<header::TbfFooterCredentialsType>()
+            + 32; // SHA256 is 32 bytes long
+                  // Length in the TLV field
+        let sha256_tlv_len = sha256_len - mem::size_of::<header::TbfHeaderTlv>();
+
+        let mut hasher = Sha256::new();
+        hasher.update(&output[0..tbfheader.binary_end_offset() as usize]);
+        let result = hasher.finalize();
+        let sha_credentials = header::TbfFooterCredentials {
+            base: header::TbfHeaderTlv {
+                tipe: header::TbfHeaderTypes::Credentials,
+                length: sha256_tlv_len as u16,
+            },
+            format: header::TbfFooterCredentialsType::SHA256,
+            data: result.to_vec(),
+        };
+        output.write_all(sha_credentials.generate().unwrap().get_ref())?;
+        footer_space_remaining -= sha256_len;
+        println!("Added SHA256 credential.");
```

Comment:
This (and others) should be gated on verbose mode.

Reply:
Done.